extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-manager.ts

```diff
@@ -167,52 +168,261 @@ export class ChePluginManager {
+     */
+    private getInstalledPluginsInfo(): { publisherName: string, version: string }[] {
+        // prepare the list of registries
+        // we need to remove the registry URI from the start of the plugin
+        const registries: string[] = [];
+        this.registryList.forEach(registry => {
+            let uri = registry.uri;
+            if (uri === this.defaultRegistry.uri) {
+                return;
+            }
+
+            if (uri.endsWith('.json')) {
+                uri = uri.substring(0, uri.lastIndexOf('/') + 1);
+            } else if (!uri.endsWith('/')) {
+                uri += '/';
+            }
+
+            registries.push(uri);
+        });
+
+        const plugins: { publisherName: string, version: string }[] = [];
+        this.installedPlugins.forEach(plugin => {
+            if (plugin.endsWith('/meta.yaml')) {
+                // it's non default registry
+                // we have to remove '/meta.yaml' from the end
+                plugin = plugin.substring(0, plugin.lastIndexOf('/'));
+            }
+
+            const version = plugin.substring(plugin.lastIndexOf('/') + 1);
+
+            // remove the version by deleting all the text withing the last '/' character
+            plugin = plugin.substring(0, plugin.lastIndexOf('/'));
+
+            // remove registry URI from the start of the plugin
+            registries.forEach(r => {
+                if (plugin.startsWith(r)) {
+                    plugin = plugin.substring(r.length);
+                }
+            });
+
+            plugins.push({
+                publisherName: plugin,
+                version
+            });
+        });
+
+        return plugins;
+    }
+
+    /**
+     * Groups all versions of the same plugin in one structure.
+     */
+    private groupPlugins(rawPlugins: ChePluginMetadata[]): ChePlugin[] {
+        const mapa: { [pluginKey: string]: ChePlugin } = {};
+
+        rawPlugins.forEach(plugin => {
+            const pluginKey = `${plugin.publisher}/${plugin.name}`;
+            let installationItem = mapa[pluginKey];
+            if (!installationItem) {
+                installationItem = {
+                    publisher: plugin.publisher,
+                    name: plugin.name,
+                    version: plugin.version,
+                    installed: false,
+                    versionList: {}
+                };
+
+                mapa[pluginKey] = installationItem;
+            } else {
+                installationItem.version = plugin.version;
+            }
+
+            installationItem.versionList[plugin.version] = plugin;
+        });
+
+        const items: ChePlugin[] = [];
+        for (const key in mapa) {
+            if (mapa.hasOwnProperty(key)) {
+                items.push(mapa[key]);
+            }
+        }
+
+        return items;
+    }
+
+    /**
+     * Installs the plugin.
+     */
+    async install(cp: ChePlugin): Promise<boolean> {
```

Comment:
Up to you, but full name is [better](https://github.com/theia-ide/theia/wiki/Coding-Guidelines#whole-words-names).

Reply:
Done